tests/periph_timer_periodic/main.c

```diff
@@ -75,11 +80,12 @@ static void cb(void *arg, int chan)
 
 static const char* _print_ok(int chan, bool *succeeded)
 {
-    if (chan == 0 && count[chan] > 0) {
-        return "OK";
+    if (chan == 0) {
+        if (count[chan] > 0) {
+            return "OK";
+        }
     }
-
-    if (chan > 0 && count[chan] == 0) {
+    else if (count[chan] == 0) {
         return "OK";
     }
```

Comment:
This is btw. cppcheck approved (I locally still run it in my editor - with appropriate include paths). The `chan > 0` looked suspiciously like a bound check to `cppcheck` and `cppecheck` pointed out that `chan < MAX_CHANNELS` would also be needed for full bounds checking.